x/fees/keeper/keeper.go

```diff
@@ -0,0 +1,69 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/params"
+	"github.com/desmos-labs/desmos/x/fees/types"
+)
+
+type Keeper struct {
+	// The reference to the ParamsStore to get and set params
+	paramSubspace params.Subspace
+	Cdc           *codec.Codec
+}
+
+func NewKeeper(cdc *codec.Codec, paramSpace params.Subspace) Keeper {
+	if !paramSpace.HasKeyTable() {
+		paramSpace = paramSpace.WithKeyTable(types.ParamKeyTable())
+	}
+
+	return Keeper{
+		Cdc:           cdc,
+		paramSubspace: paramSpace,
+	}
+}
+
+// SetParams sets params on the store
+func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
+	k.paramSubspace.SetParamSet(ctx, &params)
+}
+
+// GetParams returns the params from the store
+func (k Keeper) GetParams(ctx sdk.Context) (p types.Params) {
+	k.paramSubspace.GetParamSet(ctx, &p)
+	return p
+}
+
+// CheckFees checks whether the given fees are sufficient to pay for all the given messages.
+// The check is performed considering the minimum fee amounts specified inside the module parameters.
+func (k Keeper) CheckFees(ctx sdk.Context, fees authtypes.StdFee, msgs []sdk.Msg) error {
+	feesParams := k.GetParams(ctx)
+
+	// calculate required fees for the given messages
+	requiredFees := sdk.NewDec(0)
+	for _, msg := range msgs {
+		for _, minFee := range feesParams.MinFees {
+			if msg.Type() == minFee.MessageType {
+				requiredFees = requiredFees.Add(minFee.Amount)
+			}
+		}
+	}
+
+	stableRequiredQty := requiredFees.Mul(sdk.NewDec(1000000))
+	givenFeeAmount := sdk.NewDecFromInt(fees.Amount.AmountOf(feesParams.FeeDenom))
+
+	if !stableRequiredQty.IsZero() && stableRequiredQty.GT(givenFeeAmount) {
+		if stableRequiredQty.GT(givenFeeAmount) {
+			return sdkerrors.Wrap(sdkerrors.ErrInsufficientFee,
+				fmt.Sprintf("Expected %s %s amount, got %s", stableRequiredQty,
```

Comment:
I think this can be changed to produce messages like  `"expected at least <required_fees> but got <actual_fees> instead"